software/control/_def.py

```diff
@@ -719,14 +742,11 @@ def read_sample_formats_csv(file_path):
         print('machine-specific configuration not present, the program will exit')
         sys.exit(1)
 
-# Add this after reading the CSV files but before using DEFAULT_OBJECTIVE and WELLPLATE_FORMAT
 try:
     with open("cache/objective_and_sample_format.txt", 'r') as f:
         cached_settings = json.load(f)
-        if cached_settings.get('objective') in OBJECTIVES:
-            DEFAULT_OBJECTIVE = cached_settings['objective']
-        if cached_settings.get('wellplate_format') in WELLPLATE_FORMAT_SETTINGS or cached_settings.get('wellplate_format') == 0:
-            WELLPLATE_FORMAT = cached_settings['wellplate_format']
+        DEFAULT_OBJECTIVE = cached_settings.get('objective') if cached_settings.get('objective') in OBJECTIVES else '20x'
+        WELLPLATE_FORMAT = cached_settings.get('wellplate_format') if (cached_settings.get('wellplate_format') in WELLPLATE_FORMAT_SETTINGS or cached_settings.get('wellplate_format') == 0) else 384
```

Comment:
@sohamazing what does `cached_settings.get('wellplate_format') == 0` mean?

Reply:
if the cached wellplate format is 0, it is good and we should use 0. if the cached wellplate setting is not recognized we use the default 384.